modules/core/src/test/java/org/apache/ignite/internal/processors/cache/GridTransactionsSystemUserTimeMetricsTest.java

```diff
@@ -0,0 +1,262 @@
+import static org.apache.ignite.IgniteSystemProperties.IGNITE_LONG_TRANSACTION_TIME_DUMP_SAMPLE_LIMIT;
+import static org.apache.ignite.cache.CacheAtomicityMode.TRANSACTIONAL;
+import static org.apache.ignite.cache.CacheWriteSynchronizationMode.FULL_SYNC;
+
+/**
+ *
+ */
+public class GridTransactionsSystemUserTimeMetricsTest extends GridCommonAbstractTest {
+    /** */
+    private static final String CACHE_NAME = "test";
+
+    /** */
+    private static final String CLIENT = "client";
+
+    /** */
+    private static final String CLIENT_2 = CLIENT + "2";
+
+    /** */
+    private static final long USER_DELAY = 1000;
+
+    /** */
+    private static final long SYSTEM_DELAY = 1000;
+
+    /** */
+    private static final long LONG_TRAN_TIMEOUT = Math.min(SYSTEM_DELAY, USER_DELAY);
+
+    /** */
+    private static final long LONG_OP_TIMEOUT = 500;
+
+    /** */
+    private final LogListener logLsnr = new MessageOrderLogListener("Long transaction detected.*");
+
+    /** */
+    private static String longTranTimeoutCommon;
+
+    /** */
+    private static String longTranSampleLimit;
+
+    /** */
+    private static String longOpTimeoutCommon;
+
+    /** */
+    private volatile boolean slowPrepare;
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        ListeningTestLogger testLog = new ListeningTestLogger(false, log());
+
+        testLog.registerListener(logLsnr);
+
+        IgniteConfiguration cfg = super.getConfiguration(igniteInstanceName);
+
+        cfg.setGridLogger(testLog);
+
+        boolean isClient = igniteInstanceName.contains(CLIENT);
+
+        cfg.setClientMode(isClient);
+
+        if (!isClient) {
+            CacheConfiguration ccfg = new CacheConfiguration(CACHE_NAME);
+
+            ccfg.setAtomicityMode(TRANSACTIONAL);
+            ccfg.setBackups(1);
+            ccfg.setWriteSynchronizationMode(FULL_SYNC);
+
+            cfg.setCacheConfiguration(ccfg);
+
+        }
+
+        cfg.setCommunicationSpi(new TestCommunicationSpi());
+
+        return cfg;
+    }
+
+    /**
+     * Setting long op timeout to small value to make this tests faster
+     */
+    @Override protected void beforeTestsStarted() throws Exception {
+        super.beforeTestsStarted();
+
+        longTranTimeoutCommon = System.getProperty(IGNITE_LONG_TRANSACTION_TIME_DUMP_THRESHOLD);
+        longTranSampleLimit = System.getProperty(IGNITE_LONG_TRANSACTION_TIME_DUMP_SAMPLE_LIMIT);
+        longOpTimeoutCommon = System.getProperty(IGNITE_LONG_OPERATIONS_DUMP_TIMEOUT);
+
+        System.setProperty(IGNITE_LONG_TRANSACTION_TIME_DUMP_THRESHOLD, String.valueOf(LONG_TRAN_TIMEOUT));
+        System.setProperty(IGNITE_LONG_TRANSACTION_TIME_DUMP_SAMPLE_LIMIT, String.valueOf(1.0f));
+        System.setProperty(IGNITE_LONG_OPERATIONS_DUMP_TIMEOUT, String.valueOf(LONG_OP_TIMEOUT));
+    }
+
+    /**
+     * Returning long operations timeout to its former value.
+     */
```

Comment:
Use @WithSystemProperty for setting system property before test(s) and removing it after test(s)